### PR TITLE
fix fast reboot compatibility

### PR DESCRIPTION
**Pull request by stepanblyschak**
We should handle both cases for backward-compatible with 201803:
 - fast-reboot
 - SONIC_BOOT_TYPE=fast-reboot

Signed-off-by: Stepan Blyschak <stepanb@mellanox.com>

**NOTE** : DEPENDS on https://github.com/Azure/sonic-sairedis/pull/480

**- What I did**

**- How I did it**

**- How to verify it**

Run 201803 (157th public jenkins image + https://github.com/Azure/sonic-buildimage/pull/3077) -> 201811 (73th public jenkins image + this PR)
* verified 201811 image started in fast mode
Run 201811 (73th public jenkins image + this PR) fast reboot test
Run 201811 (73th public jenkins image + this PR) warm reboot test

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by stepanblyschak** [comment]
Retest vs please

File: files/build_templates/docker_image_ctl.j2 [hunk 1]
```diff
@@ -34,14 +34,14 @@ function updateHostName()
 function getBootType()
 {
     local BOOT_TYPE
-    case "$(cat /proc/cmdline | grep -o 'SONIC_BOOT_TYPE=\S*' | cut -d'=' -f2)" in
-    warm*)
+    case "$(cat /proc/cmdline)" in
+    *warm*)
```
Comment (qiluo-msft) [inline files/build_templates/docker_image_ctl.j2]: >warm [](start = 5, length = 4)

Be more specific `SONIC_BOOT_TYPE=warm`

File: files/build_templates/docker_image_ctl.j2 [hunk 2]
```diff
@@ -34,14 +34,14 @@ function updateHostName()
...
         TYPE='warm'
         ;;
-    fastfast)
+    *fastfast*)
```
Comment (qiluo-msft) [inline files/build_templates/docker_image_ctl.j2]: >fastfast [](start = 5, length = 8)

Be more specific `SONIC_BOOT_TYPE=fastfast`

File: files/build_templates/docker_image_ctl.j2 [hunk 3]
```diff
@@ -34,14 +34,14 @@ function updateHostName()
...
         TYPE='fastfast'
         ;;
-    fast*)
+    *fast*)
```
Comment (qiluo-msft) [inline files/build_templates/docker_image_ctl.j2]: >*fast* [](start = 4, length = 6)

This one is also too general. We can match for 2 patterns:
1. `*SONIC_BOOT_TYPE=fast*`
2. `*fast-reboot*`

File: files/build_templates/docker_image_ctl.j2 [hunk 4]
```diff
@@ -34,14 +34,14 @@ function updateHostName()
 function getBootType()
 {
     local BOOT_TYPE
```
Comment (qiluo-msft) [inline files/build_templates/docker_image_ctl.j2]: >local BOOT_TYPE [](start = 4, length = 15)

This line can be removed.

File: files/scripts/syncd.sh [hunk 5]
```diff
@@ -55,14 +55,14 @@ function wait_for_database_service()
 
 function getBootType()
```
Comment (qiluo-msft) [inline files/scripts/syncd.sh]: >getBootType [](start = 9, length = 11)

Can we reuse the function in `/usr/bin/syncd.sh`? If not, make sure you share the exactly same code snippet, add  code comments referring each other so we will not modify one without the other in future by mistake.

**Pull request merged**